tests/v2/validation/rbac/rbac_test.go

```diff
@@ -67,14 +67,19 @@ func (rb *RBTestSuite) ValidateListCluster(role string) {
 	//Testcase1 Verify cluster members - Owner/member,  Project members - Owner/member are able to list clusters
 	clusterList, err := rb.standardUserClient.Steve.SteveType(clusters.ProvisioningSteveResouceType).ListAll(nil)
 	require.NoError(rb.T(), err)
-	assert.Equal(rb.T(), 1, len(clusterList.Data))
-
 	clusterStatus := &apiV1.ClusterStatus{}
 	err = v1.ConvertToK8sType(clusterList.Data[0].Status, clusterStatus)
 	require.NoError(rb.T(), err)
 
-	actualClusterID := clusterStatus.ClusterName
-	assert.Equal(rb.T(), rb.cluster.ID, actualClusterID)
+	if role != restrictedAdmin {
+		assert.Equal(rb.T(), 1, len(clusterList.Data))
+		actualClusterID := clusterStatus.ClusterName
+		assert.Equal(rb.T(), rb.cluster.ID, actualClusterID)
+	} else {
```

Comment:
Tiny optional style nit: Consider flipping the if condition and returning early to reduce nesting and increase code readability. Handling the restrictedAdmin role at the start and then returning makes it clear what special actions are taken for this role. Would that work?

```go
	if role == restrictedAdmin {
		adminClusterList, err := rb.client.Steve.SteveType(clusters.ProvisioningSteveResouceType).ListAll(nil)
		require.NoError(rb.T(), err)
		assert.Equal(rb.T(), (len(adminClusterList.Data) - 1), len(clusterList.Data))
		return
	}

	assert.Equal(rb.T(), 1, len(clusterList.Data))
	actualClusterID := clusterStatus.ClusterName
	assert.Equal(rb.T(), rb.cluster.ID, actualClusterID)
}
```

Reply:
agreed